src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -3503,21 +3503,19 @@ public String toPlainString() {
                 return "0";
             }
             int trailingZeros = checkScaleNonZero((-(long)scale));
-            StringBuilder buf;
-            if(intCompact!=INFLATED) {
-                buf = new StringBuilder(20+trailingZeros);
-                buf.append(intCompact);
-            } else {
-                String str = intVal.toString();
-                buf = new StringBuilder(str.length()+trailingZeros);
-                buf.append(str);
-            }
-            for (int i = 0; i < trailingZeros; i++) {
-                buf.append('0');
+            String str = intCompact != INFLATED
+                ? Long.toString(intCompact)
+                : intVal.toString();
+            int len = str.length() + trailingZeros;
+            if (len < 0) {
```

Comment:
One could use `Math.addExact`, which has an intrinsic, and convert any thrown `ArithmeticException` into an OOME but I am not sure it's worth it.

Reply:
Yes, I considered that but I'm not sure how the throwing and catching of `ArithmeticException` with `*Exact` intrinsics is optimized by C2, so I have a manual overflow check for the addition.

Reply:
I think it's fine as is.